core/src/main/java/io/grpc/internal/MessageDeframer.java

```diff
@@ -311,11 +311,15 @@ private boolean readRequiredBytes() {
       while ((missingBytes = requiredLength - nextFrame.readableBytes()) > 0) {
         if (fullStreamDecompressor != null) {
           try {
-            totalBytesRead += fullStreamDecompressor.inflateBytes(missingBytes, nextFrame);
-            if (missingBytes == requiredLength - nextFrame.readableBytes()) {
+            byte[] b = new byte[Math.min(missingBytes, MAX_BUFFER_SIZE)];
```

Comment:
This may be woefully underutilized. Consider receiving 16K DATA frames for a 1 MB uncompressable message. Since this method is called every time deframe() is called, this will allocate 1 MB + (1 MB - 16 KB) + (1 MB - 32 KB) + ... + 16 KB. I'd suggest continuing to use the unused portion on subsequent reads.

Reply:
Fixed.